lib/ipam/ipam_block_reader_writer.go

```diff
@@ -145,8 +144,10 @@ func isPoolInRequestedPools(pool cnet.IPNet, requestedPools []cnet.IPNet) bool {
 	if len(requestedPools) == 0 {
 		return true
 	}
+	// Compare the requested pools against the actual pool CIDR.  Note that we don't use deep equals
+	// because golang interchangeably seems to use 4-byte and 16-byte representations of IPv4 addresses.
```

Comment:
this seems like a useful thing to know. Do you have a link or an issue you want to add to the comment so we have a reference for future?

Reply:
a golang issue?  nope - just something I've spotted in the past.  It's a bit yucky.